cmd/thanos/sidecar.go

```diff
@@ -133,8 +140,9 @@ func runSidecar(
 						"msg", "failed to fetch initial external labels. Is Prometheus running? Retrying",
 						"err", err,
 					)
+					readinessProber.SetNotReady(err)
```

Comment:
Hm, in this case it is bit unfortunate that's true. 
Being marked not ready does not cause restart of the container that would cause being not healthy. But it could cause requests being sent to the sidecar event when hasn't yet fetched the external labels. 

I'll leave just the `readinessProber.SetHealthy()` set the readiness outside of the `metricHTTPListenGroup` depending on each component.

Reply:
Moved away setting the ready status from the default http listener
https://github.com/improbable-eng/thanos/pull/656/commits/5e9a4c4b685c80d14bd3f60a93507f81a41c810b